mne/io/nsx/tests/test_nsx.py

```diff
@@ -0,0 +1,284 @@
+
+
+@requires_testing_data
+def test_stim_eog_misc_chs_in_nsx():
+    """Test stim/misc/eog channel assignments."""
+    raw = read_raw_nsx(nsx_22_fname, stim_channel="elec127", eog=["elec126"])
+    assert raw.info["chs"][127]["kind"] == FIFF.FIFFV_STIM_CH
+    assert raw.info["chs"][126]["kind"] == FIFF.FIFFV_EOG_CH
+    raw = read_raw_nsx(nsx_22_fname, stim_channel=["elec127"], eog=["elec126"])
+    assert raw.info["chs"][127]["kind"] == FIFF.FIFFV_STIM_CH
+    assert raw.info["chs"][126]["kind"] == FIFF.FIFFV_EOG_CH
+    raw = read_raw_nsx(nsx_22_fname, stim_channel=127, eog=["elec126"])
+    assert raw.info["chs"][127]["kind"] == FIFF.FIFFV_STIM_CH
+    assert raw.info["chs"][126]["kind"] == FIFF.FIFFV_EOG_CH
+    raw = read_raw_nsx(nsx_22_fname, stim_channel=[127], eog=["elec126"])
+    assert raw.info["chs"][127]["kind"] == FIFF.FIFFV_STIM_CH
+    assert raw.info["chs"][126]["kind"] == FIFF.FIFFV_EOG_CH
+    stims = [ch_info["kind"] == FIFF.FIFFV_STIM_CH for ch_info in raw.info["chs"]]
+    assert np.any(stims)
+    assert raw.info["chs"][126]["kind"] == FIFF.FIFFV_EOG_CH
+    with pytest.raises(ValueError, match="Invalid stim_channel"):
+        raw = read_raw_nsx(nsx_22_fname, stim_channel=["elec128", 129], eog=["elec126"])
+    with pytest.raises(ValueError, match="Invalid stim_channel"):
+        raw = read_raw_nsx(nsx_22_fname, stim_channel=("elec128",), eog=["elec126"])
+
+    raw = read_raw_nsx(nsx_22_fname, stim_channel="elec127", misc=["elec126", "elec1"])
+    assert raw.info["chs"][126]["kind"] == FIFF.FIFFV_MISC_CH
+    assert raw.info["chs"][1]["kind"] == FIFF.FIFFV_MISC_CH
+
+
+@requires_testing_data
+def test_nsx_ver_21():
+    """Primary tests for NEURALSG reader."""
+    with pytest.raises(NotImplementedError, match="(= NEURALSG)*not supported"):
+        read_raw_nsx(nsx_21_fname)
+
+
+@requires_testing_data
+def test_nsx():
+    """Tests for NEURALCD reader using real anonymized data."""
+    raw = read_raw_nsx(
+        nsx_test_fname,
+    )
+    assert getattr(raw, "_data", False) is False
+    assert raw.info["sfreq"] == 2000
+
+    # Check info object
+    assert raw.info["meas_date"].day == 13
+    assert raw.info["meas_date"].year == 2000
+    assert raw.info["meas_date"].month == 6
+    assert raw.info["lowpass"] == 1000
+    assert raw.info["highpass"] == 0.3
+    assert raw.info["chs"][0]["cal"] == 0.25
+    assert raw.info["chs"][0]["range"] == 1e-6
+
+    # check raw_extras
+    for r in raw._raw_extras:
+        assert r["orig_format"] == raw.orig_format
+        assert r["orig_nchan"] == 5
+        assert len(r["timestamp"]) == len(r["nb_data_points"])
+        assert len(r["timestamp"]) == len(r["offset_to_data_block"])
+
+    # Check annotations
+    assert len(raw.annotations) == 0
+
+    raw = _test_raw_reader(
+        read_raw_nsx,
+        input_fname=nsx_test_fname,
+        eog=None,
+        misc=None,
+        test_scaling=True,  # XXX this should be True
```

Comment:
The default is to test scaling and you had this set True anyway so I think this line is cruft
```suggestion
```